iis-common/src/main/java/eu/dnetlib/iis/common/report/ReportParamJsonAppender.java

```diff
@@ -0,0 +1,95 @@
+package eu.dnetlib.iis.common.report;
+
+import java.util.List;
+
+import org.apache.hadoop.util.StringUtils;
+
+import com.google.common.collect.Lists;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+import eu.dnetlib.iis.common.schemas.ReportParam;
+
+/**
+ * Appender of {@link ReportParam} into json report.
+ * 
+ * @author madryk
+ */
+public class ReportParamJsonAppender {
+
+
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Appends {@link ReportParam} into the passed json report.<br/>
+     * 
+     * The key of the report param ({@link ReportParam#getKey()}) defines
+     * where the value ({@link ReportParam#getValue()}) should be inserted
+     * in report json structure.<br/>
+     * The key is splitted by dots. Resulting values defines subsequent
+     * fields in json.<br/>
+     * For example:<br/>
+     * <code>new ReportParam("param1.paramA", "34")</code><br/>
+     * will result in following json:<br/>
+     * <code>{"param1": {"paramA": 34}}</code><br/>
+     * If report param key collides with passed json, then any
+     * conflicting node will be replaced with new report param
+     * value.
+     */
+    public void appendReportParam(JsonObject jsonReport, ReportParam reportParam) {
+        
+        String[] jsonFieldHierarchy = StringUtils.split(reportParam.getKey().toString(), '.');
+        
+        insertValue(jsonReport, Lists.newArrayList(jsonFieldHierarchy), convertParamValueToJson(reportParam));
+        
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    private JsonElement convertParamValueToJson(ReportParam reportParam) {
+        
+        long value = Long.valueOf(reportParam.getValue().toString());
```

Comment:
Now I am wondering, maybe we should keep this type aligned with `ReportParam#value` type? Either pass it as `string` or change `ReportParam#value` to `long`?

Otherwise someone generating partial report cound bump into troubles when setting `ReportParam#value` to something different than `string` representation of `float` value. Because the requirement of long type is hardcoded in this class only.